gr-pdu/python/pdu/qa_pdu_split.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+#
+# Copyright 2021 NTESS LLC.
+#
+# This file is part of GNU Radio
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+#
+
+
+from gnuradio import gr, gr_unittest, blocks, pdu
+import pmt
+import time
+
+
+class qa_pdu_split (gr_unittest.TestCase):
+
+    def setUp (self):
+        self.tb = gr.top_block ()
+
+    def tearDown (self):
+        self.tb = None
+
+    def test_001_split (self):
+        split = pdu.pdu_split()
+        d1 = blocks.message_debug()
+        d2 = blocks.message_debug()
+
+        self.tb.msg_connect((split, 'dict'), (d1, 'store'))
+        self.tb.msg_connect((split, 'data'), (d2, 'store'))
+
+        in_meta1 = pmt.dict_add(pmt.make_dict(), pmt.intern('num'), pmt.from_long(4))
+        in_meta2 = pmt.dict_add(pmt.make_dict(), pmt.intern('n'), pmt.from_long(99))
+        in_pdu = pmt.cons(in_meta1, pmt.init_u8vector(6, range(6)))
+
+        self.tb.start()
+        time.sleep(.001)
```

Comment:
Are these `time.sleep` lines here to make sure that some output is produced? @mbr0wn Added a spin wait function that would make sense here, I assume.

Reply:
@jdemel yes, this is just to give a small amount of time to ensure the message handler is called before the flowgraph is stopped. Can you point me to an example of what @mbr0wn did?

Reply:
This was merged a few days ago with #4275

Reply:
Perfect, thank you. I rebased so that is available and switched to use that, GREAT addition for message based QA @mbr0wn!